src/templates/post.js

```diff
@@ -32,6 +32,57 @@ const windowPopup = ev => {
   );
 };
 
+class TableOfContents extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      toc: [],
+    };
+  }
+
+  getTocId(element) {
+    return element.id
+      ? element.id
+      : element.textContent
+        .toLowerCase()
+        .replace(/ /g, "-")
+        .replace(/[^a-z0-9\-]/g, "");
+  }
+
+  componentDidMount() {
+    const toc = [];
+    document.querySelectorAll(".blog-post-content > h2, .blog-post-content > h3").forEach((h) => {
```

Comment:
@qmonnet this code will be executed solely on the client side, which means that very likely Google's crawler will not be able to scrape the content of the Table of Contents. I think it's a minor SEO issue. Below is the screenshot of what the browser received from the server, and what was ultimately rendered.

<img width="1600" alt="image" src="https://user-images.githubusercontent.com/58164/109285280-2d0cce80-7821-11eb-97f0-6efac4c7bca5.png">

Reply:
I see, is the ToC useful for SEO, too? :exploding_head:
Do you have an alternative suggestion? I don't know much about Gatsby, I managed to hack this ToC but I don't know how I could generate it on the server-side instead. How are we handling this for https://ebpf.io/what-is-ebpf/?

Reply:
@qmonnet really not sure if there is any serious SEO issue with TOC being absent, that is worth rewriting this part using Gatsby's internal API. So, I would suggest leaving this particular part as it is.

I mentioned this issue because in other cases, when the primary content is rendered on a page all the things should happen not on the client side, that's critically important.